File: c/resistor-color-duo/resistor_color_duo.c
#include "resistor_color_duo.h"

unsigned int color_code(const resistor_band_t bands[]) {
    unsigned int resistor_value = 0;

    // get the value of the first band
    resistor_value = bands[0] * 10;

    // get the value of the second band
    resistor_value += bands[1];

    return resistor_value;
}

File: ruby/resistor-color-duo/resistor_color_duo.rb
# frozen_string_literal: false

# https://exercism.org/tracks/ruby/exercises/resistor-color-duo
# Resistor Color Duo exercise
module ResistorColorDuo
  COLORS = %w[
    black
    brown
    red
    orange
    yellow
    green
    blue
    violet
    grey
    white
  ].freeze

  def self.color_code(color)
    COLORS.index(color)
  end

  def self.value(bands)
    resistor_value = color_code(bands[0]) * 10

    resistor_value += color_code(bands[1])

    resistor_value
  end
end
